### PR TITLE
Update CMake README.md

### PR DESCRIPTION
**Pull request by alexandraepan**
### Description 
A bunch of grammar and style fixes; add an example. 

Fixes # - _issue number(s) if exists_

- [x] - git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/oneapi-src/oneTBB/blob/master/CONTRIBUTING.md#pull-requests) for details)_

### Type of change

_Choose one or multiple, leave empty if none of the other choices apply_

_Add a respective label(s) to PR if you have permissions_

- [ ] bug fix - _change that fixes an issue_
- [ ] new feature - _change that adds functionality_
- [ ] tests - _change in tests_
- [ ] infrastructure - _change in infrastructure and CI_
- [x] documentation - _documentation update_

### Tests

- [ ] added - _required for new features and some bug fixes_
- [x] not needed

### Documentation

- [ ] updated in # - _add PR number_
- [ ] needs to be updated
- [x] not needed

### Breaks backward compatibility
- [ ] Yes
- [x] No
- [ ] Unknown

### Notify the following users
_List users with `@` to send notifications_

### CODE REVIEW

File: cmake/README.md [hunk 1]
```diff
@@ -39,19 +39,21 @@ cmake <options> <repo_root>
...
+The TBBbind library has three versions: `tbbbind`, `tbbbind_2_0`, and `tbbbind_2_5`. Each of these versions is linked with the corresponding HWLOC* library version: 
+- `tbbbind` links with `HWLOC 1.11.x`
+- `tbbbind_2_0` links with `HWLOC 2.1–2.4`
+- `tbbbind_2_5` links with `HWLOC 2.5` and later
 
 The search for a suitable version of the HWLOC library is enabled by default. If you want to use a specific version of the library, you can specify the path to it manually using the following CMake variables:
 
  - `CMAKE_HWLOC_<HWLOC_VER>_LIBRARY_PATH` - path to the corresponding HWLOC version shared library on Linux* OS or path to `.lib` file on Windows* OS
- - `CMAKE_HWLOC_<HWLOC_VER>_INCLUDE_PATH` - path to the corresponding HWLOC version including directory
+ - `CMAKE_HWLOC_<HWLOC_VER>_INCLUDE_PATH` - path to the corresponding HWLOC version, including directory
```
Comment (dnmokhov) [inline cmake/README.md:56]: Not sure of the right wording, but this is the path of the HWLOC/include directory, not of the HWLOC directory.

Comment (alexandraepan) [reply cmake/README.md:56]: thanks, fixed!

File: cmake/README.md [hunk 2]
```diff
@@ -223,14 +240,14 @@ target_link_libraries(foo TBB::tbb)
 oneTBB components can be passed to [find_package](https://cmake.org/cmake/help/latest/command/find_package.html) after keyword ``COMPONENTS`` or ``REQUIRED``.
 Use basic names of components (`tbb`, `tbbmalloc`, etc.).
 
-If components are not specified then the default set is used: `tbb`, `tbbmalloc` and ``tbbmalloc_proxy``.
+If components are not specified, then the default setting is used: `tbb`, `tbbmalloc`, and ``tbbmalloc_proxy``.
```
Comment (omalyshe) [inline cmake/README.md:243]: Here "set of components" was meant.

File: cmake/README.md [hunk 3]
```diff
@@ -39,19 +39,21 @@ cmake <options> <repo_root>
...
+The TBBbind library has three versions: `tbbbind`, `tbbbind_2_0`, and `tbbbind_2_5`. Each of these versions is linked with the corresponding HWLOC* library version: 
+- `tbbbind` links with `HWLOC 1.11.x`
+- `tbbbind_2_0` links with `HWLOC 2.1–2.4`
+- `tbbbind_2_5` links with `HWLOC 2.5` and later
 
 The search for a suitable version of the HWLOC library is enabled by default. If you want to use a specific version of the library, you can specify the path to it manually using the following CMake variables:
 
  - `CMAKE_HWLOC_<HWLOC_VER>_LIBRARY_PATH` - path to the corresponding HWLOC version shared library on Linux* OS or path to `.lib` file on Windows* OS
- - `CMAKE_HWLOC_<HWLOC_VER>_INCLUDE_PATH` - path to the corresponding HWLOC version including directory
+ - `CMAKE_HWLOC_<HWLOC_VER>_INCLUDE_PATH` - path to the corresponding HWLOC-version including directory
```
Comment (omalyshe) [inline cmake/README.md:56]: Looks like the dash is not needed here

File: cmake/README.md (hunk 2 above)
Comment (alexandraepan) [reply cmake/README.md]: fixed

File: cmake/README.md (hunk 3 above)
Comment (alexandraepan) [reply cmake/README.md]: fixed

**Review by omalyshe — APPROVED** [review]
LGTM.

**Pull request merged**